File: TypeScript/2220.ts
/**
 * 2220. Minimum Bit Flips to Convert Number
 * https://leetcode.com/problems/minimum-bit-flips-to-convert-number/
 */

function minBitFlips(start: number, goal: number): number {
  const countOnes = (n) => {
    return n === 0 ? n : (n & 1) + countOnes((n >>= 1));
  };

  return countOnes(start ^ goal);
}

File: Rust/2220.rs
/**
 * 2220. Minimum Bit Flips to Convert Number
 * https://leetcode.com/problems/minimum-bit-flips-to-convert-number/
 */

impl Solution {
  pub fn min_bit_flips(start: i32, goal: i32) -> i32 {
    return (start ^ goal).count_ones() as i32;
  }
}
